### DISCUSSION TITLE
Next.js protected API routes with Prisma & next-auth in edge runtime

### DISCUSSION THREAD

**Discussion by segersniels** [prisma/prisma#22529] [Q&A] [answered]
### Question

Unfortunately it seems like the current prisma client (even with v5) does not have support for edge runtime yet. I would love to protect my API routes to see if a user has a valid subscription before accessing specific API routes.

Does anyone have any ideas how to implement something if I can't access the session? I was thinking of requiring the client to send their session to the API route and then checking there. But any tech savvy person can find his way around and spoof the request.

**Comment by segersniels** [comment]
There is this current discussion https://github.com/prisma/prisma/issues/18763#issuecomment-1841757630 about possible edge runtime support. But nothing publicly released yet.

**Comment by segersniels** [comment]
I was potentially thinking of making a JWT token that expires in a route handler that needs to be passed with the request and verified to be deemed a successful authorization. But I ran into `crypto` not being available on the server which `jsonwebtoken` needed.

Any other ideas?

**Reply by jonahallibone** [reply]
[Jose](https://www.npmjs.com/package/jose) works in edge runtimes. Give that a shot.

**Reply by segersniels** [reply, marked as answer]
I eventually just worked around it by forcing the `jwt` strategy alongside the adapter and exposing everything from the token onto the session through the `callbacks` config while an adapter-less instance to access the session in the edge runtime.  Solutions seems sufficient for now.

**Reply by yvmiko** [reply]
This also worked for me, hoping it doesn't break in production.
